wallet/src/repository/wallet/tests/factories.rs

```diff
@@ -3,7 +3,10 @@ use witnet_data_structures::chain::Hash;
 use super::*;
 use crate::db::HashMapDb;
 
-pub fn wallet(data: Option<HashMapDb>) -> (Wallet<db::HashMapDb>, db::HashMapDb) {
+pub fn wallet(
+    data: Option<HashMapDb>,
+    store_master_key: bool,
+) -> (Wallet<db::HashMapDb>, db::HashMapDb) {
```

Comment:
This looks like a hack that's only needed because of one test. In some other tests we solve this by allowing to mutate the input, for example in `test_block` in `validations/src/tests.rs`. So a more future-proof approach would be to accept a closure here:

```suggestion
    f: F,
) -> (Wallet<db::HashMapDb>, db::HashMapDb)
where F: FnOnce(&mut Wallet<db::HashMapDb>)
{
```

But this would require all tests to pass an empty `||` as an argument, which is not nice.

So I think a better approach would be to keep a `wallet` function with only one argument, and call a different function like "wallet2", which accepts two arguments, like this:

```rust
pub fn wallet(
// ...
{
    wallet2(data, true)
}
```

This is the hack we use to emulate default arguments in Rust. In this case, we default `store_master_key = true`. This way all the tests remain unchanged, and the one test that needs `store_master_key = false` can simply call `wallet2` (or `wallet_inner` or `wallet_with_args` or whatever).

Reply:
good

Reply:
I implemented the latter